GitUI/Avatars/StaticImageAvatarProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+using System.Drawing;
+using System.Threading.Tasks;
+
+namespace GitUI.Avatars
+{
+    public sealed class StaticImageAvatarProvider : IAvatarProvider
+    {
+        private readonly Image _image;
+        private readonly Dictionary<int, Image> _sizeCache = new Dictionary<int, Image>();
+
+        public StaticImageAvatarProvider(Image image)
+        {
+            _image = image;
+            _sizeCache.Add(_image.Height, image);
+        }
+
+        /// <inheritdoc />
+        public Task<Image> GetAvatarAsync(string email, string name, int imageSize)
+        {
+            return Task.FromResult(GetCachedResizedImage(imageSize));
+        }
```

Comment:
Why not async?

Reply:
I reused a lot of existing (and field tested code) and it wasn't async before. I can certainly give it a go if you want me to change that.